pds4indextools/pds4_create_xml_index.py

```diff
@@ -1475,8 +1484,12 @@ def main(cmd_line=None):
 
         if args.generate_label[0] == 'Product_Ancillary':
             label_content['Product_Ancillary'] = True
-        else:
+        elif args.generate_label[0] == 'Product_Metadata_Supplemental':
```

Comment:
I think this argument check should be done as part of argparse, like you validate the extra file info options. Right now you could spend a long time scraping thousands of labels and then only at the end get told your label option was incorrect. So create an appropriate "validate" function and pass it to the argument definition.